lib/player.js

```diff
@@ -3818,8 +3818,6 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
           shaka.util.Error.Code.CANNOT_ADD_EXTERNAL_TEXT_TO_LIVE_STREAM);
     }
 
-    let mimeType = mime;
-
     if (!mimeType) {
       // Try using the uri extension.
       const extension = shaka.media.ManifestParser.getExtension(uri);
```

Comment:
If MIME type is not given and can't be deduced from the file extension, perhaps we should throw a specific error code instead of silently creating a track that won't be associated with any parser.

Reply:
I am going to throw an exception.
I know it's not ideal, but next week (when these two PRs merge) I'm going to launch an PR for addTextTrack to return a promise and we can make a network request to get the mimeType if needed, and also to be able to add specific error codes.